### ISSUE TITLE
[slider] Focus styling for slider thumb and the hidden `<input />`

### ISSUE THREAD

**Issue by vladmoroz** [mui/base-ui#2512]
# Get help

Is it intended that slider thumb is two focusable elements? The wrapper gets focused when tabbing into the slider via keyboard, but the hidden input in the thumb gets focused when moving the slider via mouse.

https://github.com/user-attachments/assets/13d2ec4e-1601-4dfe-b8ef-f12bb1c05987

https://base-ui.com/react/components/slider

On the video, I'm interacting with the pointer first and then continuing with the keyboard, which fails to show the `:focus-visible` styles because under the hood, the hidden input is focused.

I'm confused if it's a mistake or if I'm supposed to style both elements.

**Comment by vladmoroz** [comment]
Related, but we are doing some manual focus management and having an `<input />` that catches events which we don't have immediate access to is proving high-friction for other reasons.

Before migrating from Radix, I could just attach a data attribute to the focused element:

<img width="449" height="204" alt="Image" src="https://github.com/user-attachments/assets/ad3b630b-39eb-4efc-b1a1-74e8a662804e" />
<br />
<br />

With Base UI, to add a data attribute I need to pull out the input from the ref and make sure to merge refs:

<img width="640" height="407" alt="Image" src="https://github.com/user-attachments/assets/ea30fb69-74df-4737-9340-b4e913d535ff" />

**Comment by mj12albert** [comment]
> Is it intended that slider thumb is two focusable elements?

Possibly for legacy reasons, but I think you definitely shouldn't have to style the `input` elements at all 😓 

Different `input` elements were added in https://github.com/mui/base-ui/pull/1929 for form submission. Maybe the `input` inside the Thumb is redundant now, if so it would be great to get rid of a lot of one-off `render` prop workarounds that were enabling it.

**Comment by vladmoroz** [comment]
Could Thumb render the range input without a wrapper?

The range input itself doesn't bother me, the problem is that there are two focusable elements I need to style focus on and that I don't have direct access to the range input DOM node.

**Comment by mj12albert** [comment]
Yeah it would be good if there was just one element here, but was thinking that a `div role="slider"` be more convenient to style than `input type="range"` @vladmoroz

**Comment by oliviertassinari** [comment]
The a11y of this component seems a bit buggy:

- [ ] 1. When using the keyboard modality, the focus-visible style doesn't show:

https://github.com/user-attachments/assets/b39a7e37-c1e3-468e-88c1-737065e36050

Same bug in https://joy-ui.com/slider/. But in https://react-spectrum.adobe.com/react-aria/Slider.html or https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/input/range it's correct.

- [ ] 2. When you focus the slider with a keyboard (focusing the div), and you change the value, VoiceOver doesn't announce value changes. Probably same bug as 1.
- [ ] 3. #2545
- [ ] 4. There is a `role: 'group'`, I don't understand why it's here for a single slider. VoiceOver announces it. Is this for the range slider only?

**Comment by mj12albert** [comment]
> Is it intended that slider thumb is two focusable elements?

We found it's intentional that the `input` must receive focus in order to support Android Talkback: https://github.com/mui/base-ui/pull/2537#issuecomment-3218335240

RA's slider ([sandbox](https://codesandbox.io/p/sandbox/react-aria-slider-forked-3l4rmy)) which seems to have the most correct behavior, also has these issues:
- the `input` element nested in the thumb `div` catches focus/blur, but the `div` gets all other events
- a custom data attribute passed to `SliderThumb` is set on the `div`, not the `input`

Additionally RA polyfills `:focus-visible` to avoid involving the `input` in styling @vladmoroz
